File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/DTOS/UpdateProductDTO.java
package app.aurafitbackend.DTOS;

import app.aurafitbackend.Beans.Category;
import app.aurafitbackend.Beans.ProductType;
import app.aurafitbackend.Enums.Gender;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
public class UpdateProductDTO {

    private Long id;
    private String name;
    private String description;
    private Gender gender;
    private Category category;
    private ProductType productType;

}

File: aurafit-frontend/src/Models/DTOS/UpdateProductDTO.ts
import {Gender} from "../Enums/Gender.ts";
import {Category} from "../Category.ts";
import {ProductType} from "../ProductType.ts";

export class UpdateProductDTO {
    id: number
    name: string
    description: string
    gender: Gender
    category: Category
    productType: ProductType

    constructor(id: number, name: string, description: string, gender: Gender, category: Category, productType: ProductType) {
        this.id = id
        this.name = name
        this.description = description
        this.gender = gender
        this.category = category
        this.productType = productType
    }
}